crates/interpreter/src/interpreter/contract.rs

```diff
@@ -19,8 +19,16 @@ pub struct Contract {
 }
 
 impl Contract {
-    pub fn new(input: Bytes, bytecode: Bytecode, address: B160, caller: B160, value: U256) -> Self {
-        let bytecode = to_analysed(bytecode).try_into().expect("it is analyzed");
+    pub fn new(
+        input: Bytes,
+        bytecode: &Bytecode,
+        address: B160,
+        caller: B160,
+        value: U256,
+    ) -> Self {
+        let bytecode = to_analysed(bytecode.clone())
```

Comment:
I would remove the clone here, and instantiate the Bytecode without `Box`, the clone seems redundant.